### PR TITLE
Fix error with bam input

### PR DESCRIPTION
**Pull request by pimarin**
Previous version of polypolish failed to deal with `.bam` file, due to an error in a `if/else` structure which do not test bam input.
I added it and also bam tests

### CODE REVIEW

File: tools/polypolish/polypolish.xml [hunk 1]
```diff
@@ -14,25 +14,31 @@
                     For single SAM
         ======================================*#
         #if $input.sam_data_type.sam_selector == 'single'
-            #if $input.sam_data_type.single_sam.ext == 'unsorted.bam'
+            #if $input.sam_data_type.single_sam.ext in ("unsorted.bam", "bam")
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:17]:
```suggestion
            #if $input.sam_data_type.single_sam.is_of_type("unsorted.bam")
```
which includes all flavors and sort orders of bam files.

File: tools/polypolish/polypolish.xml [hunk 2]
```diff
@@ -51,10 +57,12 @@
         #elif $input.sam_data_type.sam_selector == 'multiple_single'
             mkdir single_collection &&
             #for $value, $single_sam in enumerate($input.sam_data_type.single_collection):
-                #if $single_sam.ext == 'unsorted.bam'
+                #if $single_sam.ext in ("unsorted.bam", "bam")
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:60]:
```suggestion
                #if $single_sam.is_of_type("unsorted.bam")
```

File: tools/polypolish/polypolish.xml [hunk 3]
```diff
@@ -118,33 +126,33 @@
                     <option value="multiple_paired">Multiple paired SAM/BAM files</option>
                 </param>
                 <when value="single">
-                    <param name="single_sam" type="data" format="sam,unsorted.bam" label="Select a SAM/BAM file" help="Specify dataset with only one SAM/BAM file"/>
+                    <param name="single_sam" type="data" format="sam,unsorted.bam,bam" label="Select a SAM/BAM file" help="Specify dataset with only one SAM/BAM file"/>
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:129]: bam does not make sense as an addition here, that is included with `unsorted.bam` which refers to the base class
```suggestion
                    <param name="single_sam" type="data" format="sam,unsorted.bam" label="Select a SAM/BAM file" help="Specify dataset with only one SAM/BAM file"/>
```

File: tools/polypolish/polypolish.xml [hunk 4]
```diff
@@ -118,33 +126,33 @@
...
                 </when>
                 <when value="paired">
-                    <param name="R1_sam" type="data" format="sam,unsorted.bam" label="Select forward SAM/BAM file" help="Specify the forward SAM/BAM files"/>
-                    <param name="R2_sam" type="data" format="sam,unsorted.bam" label="Select reverse SAM/BAM file" help="Specify the reverse SAM/BAM files"/>
+                    <param name="R1_sam" type="data" format="sam,unsorted.bam,bam" label="Select forward SAM/BAM file" help="Specify the forward SAM/BAM files"/>
+                    <param name="R2_sam" type="data" format="sam,unsorted.bam,bam" label="Select reverse SAM/BAM file" help="Specify the reverse SAM/BAM files"/>
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:133]:
```suggestion
                    <param name="R1_sam" type="data" format="sam,unsorted.bam" label="Select forward SAM/BAM file" help="Specify the forward SAM/BAM files"/>
                    <param name="R2_sam" type="data" format="sam,unsorted.bam" label="Select reverse SAM/BAM file" help="Specify the reverse SAM/BAM files"/>
```

File: tools/polypolish/polypolish.xml [hunk 5]
```diff
@@ -118,33 +126,33 @@
...
                     <expand macro="filter_option"/>
                 </when>
                 <when value="multiple_single">
-                    <param name="single_collection" format="sam,unsorted.bam" type="data_collection" collection_type="list" label="Single-end collection" help="Specify a list of single-end dataset"/>
+                    <param name="single_collection" format="sam,unsorted.bam,bam" type="data_collection" collection_type="list" label="Single-end collection" help="Specify a list of single-end dataset"/>
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:137]:
```suggestion
                    <param name="single_collection" format="sam,unsorted.bam" type="data_collection" collection_type="list" label="Single-end collection" help="Specify a list of single-end dataset"/>
```

File: tools/polypolish/polypolish.xml [hunk 6]
```diff
@@ -118,33 +126,33 @@
...
                 </when>
                 <when value="multiple_paired">
-                    <param name="paired_collection" format="sam,unsorted.bam" type="data_collection" collection_type="list:paired" label="Single-end collection" help="Specify a list of single-end dataset"/>
+                    <param name="paired_collection" format="sam,unsorted.bam,bam" type="data_collection" collection_type="list:paired" label="Single-end collection" help="Specify a list of single-end dataset"/>
```
Comment (mvdbeek) [inline tools/polypolish/polypolish.xml:140]:
```suggestion
                    <param name="paired_collection" format="sam,unsorted.bam" type="data_collection" collection_type="list:paired" label="Single-end collection" help="Specify a list of single-end dataset"/>
```

**Comment by pimarin** [comment]
It seem to be okay now, if you want to review @mvdbeek.

**Review by bernt-matthias — APPROVED** [review]
Looks good to me.

**Pull request merged**